src/components/FormHelpMessage.js

```diff
@@ -35,14 +35,24 @@ const FormHelpMessage = (props) => {
         return null;
     }
 
+    const textStyles = [props.isError ? styles.formError : styles.formHelp, styles.mb0];
+
     return (
         <View style={[styles.flexRow, styles.alignItemsCenter, styles.mt2, styles.mb1, ...props.style]}>
             {props.isError && <Icon src={Expensicons.DotIndicator} fill={colors.red} />}
             <View style={[styles.flex1, styles.ml2]}>
                 {props.children || (
-                    <Text style={[props.isError ? styles.formError : styles.formHelp, styles.mb0]}>
-                        {props.message}
-                    </Text>
+                    <>
+                        {_.isString(props.message) ? (
+                            <Text style={textStyles}>
+                                {props.message}
+                            </Text>
+                        ) : _.map(props.message, msg => (
+                            <Text key={msg} style={textStyles}>
+                                {msg}
+                            </Text>
```

Comment:
I like this rendering approach, but I feel like the "message" prop isn't the right name anymore. Additionally it feels like we're starting to stretch this component beyond it's original purpose.

In the Form we pass errorText to the inputs, but I also think that name is invalid at this point. How about within the Form component we construct errorText so that it's a string with line breaks between each error message?

Something like this, I tested it quickly in the FormHelpMessage component and it appears works well, but it would be better to do this in the Form itself.
```javascript
    const message = _.isString(props.message) ? props.message : _.reduce(props.message, (totalMessage, curMessage) => `${totalMessage}\n${curMessage}`, '');
```

Then the errorText and message props can remain a strings and less changes are needed.

Reply:
+1